examples/host_func_test.go

```diff
@@ -19,22 +22,22 @@ type testKey struct{}
 var hostFuncWasm []byte
 
 func Test_hostFunc(t *testing.T) {
-	allocateBuffer := func(context.Context, uint32) uint32 {
-		panic("unimplemented")
-	}
+	// The function for allocating the in-Wasm memory region.
+	// We resolve this function after main module instantiation.
+	var allocateInWasmBufferFn wasm.Function
+
+	var expectedBase64String string
 
 	// Host-side implementation of get_random_string on Wasm import.
-	getRandomString := func(ctx wasm.ModuleContext, retBufPtr uint32, retBufSize uint32) {
+	getRandomBytes := func(ctx wasm.ModuleContext, retBufPtr uint32, retBufSize uint32) {
 		// Assert that context values passed in from CallFunctionContext are accessible.
 		contextValue := ctx.Context().Value(testKey{}).(int64)
 		require.Equal(t, int64(12345), contextValue)
 
-		const bufferSize = 10000 // force memory space grow to ensure eager failures on missing setup
-		// Allocate the in-Wasm memory region so we can store the generated string.
-		// Note that this is recursive call. That means that this is the VM function call during the VM function call.
-		// More precisely, we call test.base64 (in Wasm), and the function in turn calls this get_random_string function,
-		// and we call test.allocate_buffer (in Wasm) here: host->vm->host->vm.
-		offset := allocateBuffer(ctx.Context(), bufferSize)
+		const bufferSize = 10
+		res, err := allocateInWasmBufferFn.Call(ctx.Context(), uint64(bufferSize))
```

Comment:
Aside from the Adiran's discussion on showing how to stub, if we don't show such stub and just show calling a wasm function here, then getting the function by `ctx.Function("allocate_buffer")` inside `getRandomBytes` would be current simpler API.

Reply:
yep agreed. there's no point to play "the shell game" unless we are trying to show how to unwind deps.